test/unit/proxy_config_test.rb

```diff
@@ -113,6 +113,17 @@ def test_create_without_service_token
     assert proxy_config.valid?
   end
 
+  test 'missing api backend when service_mesh' do
+    service = FactoryBot.create(:simple_service)
+    proxy = FactoryBot.create(:proxy, service: service)
```

Comment:
```suggestion
    proxy = service.proxy
```

The factory already creates it